### ISSUE TITLE
AD410 no longer supported?

### ISSUE THREAD

**Issue by nicolasnoble** [rroller/dahua#134]
This may or may not be an issue with this dahua integration directly, so bear with me as I explain what's going on here.

I've purchased an AD410 a few days ago with the hope I'd be able to integrate it with my Home Assistant, but it turns out it's now impossible. Basically, it looks like Amcrest pushed a new firmware to the AD410 (the 1.000.0000000.7.R.210220), and my device came with it pre-loaded, which fully disables the web interface. It may or may not be a mistake as the way it's "disabled" is that the file `js/merge.js` referenced from the HTML does not exist and returns a 404. The doorbell otherwise works perfectly from Amcrest's Android application. As a result, one can no longer authenticate directly to it, or set a password. When trying to add the doorbell to Home Assistant through this integration, and using the default "admin"/"admin" login and password, but it doesn't work:

![image](https://user-images.githubusercontent.com/7281574/147903576-8069f8ef-937b-4bf0-addd-327d32c5535b.png)

Is this a known issue? Are there workarounds?

Additionally, could someone dump the javascript from an older version of the firmware so I can try side loading it in the browser and set the account? Or maybe tell me which endpoint is being used to set the password initially, and potentially bypass the whole web ui and use the doorbell directly through home assistant?

Thank you

**Comment by rroller** [comment]
This is the first time I'm hearing of this. I'll do some digging

**Comment by nicolasnoble** [comment]
Thanks. I don't mind exposing the camera to you through a VPN or an SSH tunnel if you want to see for yourself.

**Comment by rroller** [comment]
I have one of these. Worst case is I'll upgrade it.

**Comment by nicolasnoble** [comment]
Maybe make sure you dump the web ui first :-) I haven't seen how to downgrade the firmware...

**Comment by rroller** [comment]
How do you access the webui? I didn't know these had one?

**Comment by nicolasnoble** [comment]
Basically this:

https://support.amcrest.com/hc/en-us/articles/360002013712-How-To-Access-Your-Web-UI

But, now I'm confused: how did you initially set up your AD410 using home assistant without accessing the webui? The instructions seems to indicate one would log in using the login and password set through the webui.

**Comment by rroller** [comment]
I just used the iOS app

**Comment by nicolasnoble** [comment]
So the amcrest Android app requires you to create an account on their servers in order to access the device. Are you using those login and passwords? For me the login is my full email address for instance. Or is the iOS application letting you create a local account on the device?

Let me try with my email and password...

**Comment by nicolasnoble** [comment]
Nope, the email and password I'm using for the Android application isn't letting me connect to the doorbell locally. So there has to be something you've done in the iOS app that sets up a local account on the device.

**Comment by rroller** [comment]
There's a local password on the device itself. Look on the back of it

**Comment by nicolasnoble** [comment]
Oh my, this is totally unintuitive, thanks. All of the documentation from Amcrest says to use the webui and that the default password is "admin".

https://support.amcrest.com/hc/en-us/articles/4403757995917-How-to-Factory-Reset-Your-Amcrest-Doorbell-AD410-

Thank you. It's a bit late for me to go rip off my doorbell and get its sticker on the back, but I'll try it first thing tomorrow.

**Comment by rroller** [comment]
Wait let me confirm the password is on the back. I have a lot of devices and I might be getting them confused

**Comment by rroller** [comment]
I think I used the app to reset the local password. See this screenshot 


![image](https://user-images.githubusercontent.com/445655/147909399-9ee52d01-a754-413e-9b68-0daca33ecd32.jpeg)

And this is the firmware version I'm on 

![image](https://user-images.githubusercontent.com/445655/147909439-38342758-c165-4a56-a893-3ae59a995c29.jpeg)

**Comment by nicolasnoble** [comment]
Okay, I'm losing my mind here, as I totally managed to get it to work with sheer luck. The "set password" from your screen asks for the "old device password", but I've never set one. I tried "admin", no luck. I tried the password for my amcrest account, and it worked.

So long story short, I finally just managed to add my AD410 to home assistant by using the login "admin", and the password I just managed to set in the app, by using my cloud password to change it.

That's totally ridiculous and absolutely undocumented by Amcrest, and also different from their other products.

Thank you so much for bearing with me on this one, and guiding me through finding the actual problem.

TL;DR: there is no web ui at all, despite amcrest claiming there is. The default device password isn't "admin", despite amcrest's documentation, it's the same password as your cloud account. But the username is still "admin".

**Comment by rroller** [comment]
Nice! FYI - I block all my devices from the internet after I set them up.

**Comment by nicolasnoble** [comment]
> Nice! FYI - I block all my devices from the internet after I set them up.

Yeah that's usually my go-to move too :-)

**Comment by nicolasnoble** [comment]
Just as an FYI, I had a very lovely chat with Amcrest's support: basically, the device is intended to be cloud-only, so the local API stuff is likely considered a bug, and not a feature. I wouldn't be shocked if it's fully disabled in a future firmware release.
